### PR TITLE
Create note and Edit note

### PR DESCRIPTION
**Pull request by KaustubhM1997**

### CODE REVIEW

**Review by Shrinkhlarajpoot — APPROVED** [review]
Clean code... good to merge

File: src/components/EditNotes/editnote.css [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  height: 100%;
+  top: 0;
+  right: 0;
+  background-color: rgba(0, 0, 0, 0.3);
+  z-index: 9;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  /* backdrop-filter: blur(1px); */
+  /* opacity: 0.3; */
```
Comment (srishti-maurya) [inline src/components/EditNotes/editnote.css]: you may delete the commented code

File: src/components/EditNotes/editnote.jsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  const { currentEditNote } = notesState;
+
+  //The current edit note is nothing but the modal that pops up
+
+  const textInputRef = useRef();
+
+  const editSubmitFormHandler = (e) => {
+    e.preventDefault();
+    console.log(currentEditNote, "done");
```
Comment (srishti-maurya) [inline src/components/EditNotes/editnote.jsx]: you may remove `console.log`

**Review by srishti-maurya — APPROVED** [review]
I hope the 2 files that are commented would be used in the future. Also, you may remove `console.log` from the codes, and use `Link` from `react-router-dom` instead of `anchor` tags.
I liked the way you have explained the codes via comments

File: src/components/EditNotes/editnote.css [hunk 3]
```diff
@@ -0,0 +1,22 @@
+.edit-note {
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  right: 0;
+  background-color: rgba(0, 0, 0, 0.3);
```
Comment (manojasarsa) [inline src/components/EditNotes/editnote.css]: you can make this color as a variable and then use it.

File: src/components/NoteCard/notecard.css [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  flex-direction: column;
+}
+
+.note-card-title-pin {
+  display: flex;
+  justify-content: space-between;
+}
+
+.main-note-card-wrapper:hover {
+  box-shadow: var(--full-box-shadow);
```
Comment (manojasarsa) [inline src/components/NoteCard/notecard.css:32]: nice use of variable

File: src/contexts/notes-context.jsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+            console.log("cominh");
+            dispatchNotes({
+              type: "SET_NOTES",
+              payload: {
+                notesList: data.notes,
+              },
+            });
+          }
+        } catch (err) {
+          console.log(err);
```
Comment (manojasarsa) [inline src/contexts/notes-context.jsx:133]: Use `console.error` for handling errors

**Review by manojasarsa — APPROVED** [review]
Rest, looks clean and well structured.

**Pull request merged**